render the function `error' unambiguous
--- a/log.h
+++ b/log.h
@@ -15,6 +15,11 @@
 #ifndef SSH_LOG_H
 #define SSH_LOG_H
 
+#ifdef error
+#undef error
+#endif
+#define error unambigous_log_error
+
 /* Supported syslog facilities and levels. */
 typedef enum {
 	SYSLOG_FACILITY_DAEMON,
@@ -57,7 +62,7 @@
 
 void     fatal(const char *, ...) __attribute__((noreturn))
     __attribute__((format(printf, 1, 2)));
-void     error(const char *, ...) __attribute__((format(printf, 1, 2)));
+void     unambigous_log_error(const char *, ...) __attribute__((format(printf, 1, 2)));
 void     sigdie(const char *, ...)  __attribute__((noreturn))
     __attribute__((format(printf, 1, 2)));
 void     logit(const char *, ...) __attribute__((format(printf, 1, 2)));
--- a/log.c
+++ b/log.c
@@ -155,7 +155,7 @@
 /* Error messages that should be logged. */
 
 void
-error(const char *fmt,...)
+unambigous_log_error(const char *fmt,...)
 {
 	va_list args;
 
